sklearn/utils/_param_validation.py

```diff
@@ -415,31 +421,137 @@ def __str__(self):
         )
 
 
-def generate_invalid_param_val(constraint):
+def generate_invalid_param_val(constraint, constraints=None):
     """Return a value that does not satisfy the constraint.
 
+    Raises a NotImplementedError if there exists no invalid value for this constraint.
+
     This is only useful for testing purpose.
 
     Parameters
     ----------
     constraint : Constraint
         The constraint to generate a value for.
 
+    constraints : list of Constraint or None, default=None
+        The list of all constraints for this parameter. If None, the list containt only
+        the constraint is used.
+
     Returns
     -------
     val : object
         A value that does not satisfy the constraint.
     """
     if isinstance(constraint, StrOptions):
         return f"not {' or '.join(constraint.options)}"
-    elif isinstance(constraint, Interval):
-        interval = constraint
+
+    if not isinstance(constraint, Interval):
+        raise NotImplementedError
+
+    # constraint is an interval
+    constraints = [constraint] if constraints is None else constraints
+    return _generate_invalid_param_val_interval(constraint, constraints)
+
+
+def _generate_invalid_param_val_interval(interval, constraints):
+    """Return a value that does not satisfy an interval constraint.
+
+    Generating an invalid value for an integer interval depends on the other constraints
+    since an int is a real, meaning that it can be valid for a real interval.
+    Assumes that there can be at most 2 interval constraints: one integer interval
+    and/or one real interval.
+
+    This is only useful for testing purpose.
+
+    Parameters
+    ----------
+    interval : Interval
```

Comment:
I think that we should specify that they are instances of Interval, Constraint, etc.

We can only correct the code touch now and make a small subsequent PR touching only the docstrings.